src/VisualStudio/CSharp/Test/UnifiedSettings/CSharpUnifiedSettingsTests.cs

```diff
@@ -0,0 +1,62 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.IO;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+using Microsoft.CodeAnalysis.Editor.CSharp.CompleteStatement;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.VisualStudio.LanguageServices;
+using Microsoft.VisualStudio.LanguageServices.UnitTests.UnifiedSettings;
+using Newtonsoft.Json.Linq;
+using Xunit;
+
+namespace Roslyn.VisualStudio.CSharp.UnitTests.UnifiedSettings
+{
+    public class CSharpUnifiedSettingsTests : UnifiedSettingsTests
+    {
+        internal override ImmutableArray<IOption2> OnboardedOptions => ImmutableArray.Create<IOption2>(
+            CompletionOptionsStorage.TriggerOnTypingLetters,
+            CompletionOptionsStorage.TriggerOnDeletion,
+            CompletionOptionsStorage.TriggerInArgumentLists,
+            CompletionViewOptionsStorage.HighlightMatchingPortionsOfCompletionListItems,
+            CompletionViewOptionsStorage.ShowCompletionItemFilters,
+            CompleteStatementOptionsStorage.AutomaticallyCompleteStatementOnSemicolon,
+            CompletionOptionsStorage.SnippetsBehavior,
+            CompletionOptionsStorage.EnterKeyBehavior,
+            CompletionOptionsStorage.ShowNameSuggestions,
+            CompletionOptionsStorage.ShowItemsFromUnimportedNamespaces,
+            CompletionViewOptionsStorage.EnableArgumentCompletionSnippets,
+            CompletionOptionsStorage.ShowNewSnippetExperienceUserOption
+        );
+
+        internal override ImmutableDictionary<IOption2, object> OptionsToDefaultValue => ImmutableDictionary<IOption2, object>.Empty.
+            Add(CompletionOptionsStorage.SnippetsBehavior, SnippetsRule.AlwaysInclude).
+            Add(CompletionOptionsStorage.EnterKeyBehavior, EnterKeyRule.Never).
+            Add(CompletionOptionsStorage.TriggerOnDeletion, false).
+            Add(CompletionOptionsStorage.ShowItemsFromUnimportedNamespaces, true).
+            Add(CompletionViewOptionsStorage.EnableArgumentCompletionSnippets, false).
+            Add(CompletionOptionsStorage.ShowNewSnippetExperienceUserOption, false);
+
+        internal override ImmutableDictionary<IOption2, ImmutableArray<object>> EnumOptionsToValues => ImmutableDictionary<IOption2, ImmutableArray<object>>.Empty.
+                Add(CompletionOptionsStorage.SnippetsBehavior, ImmutableArray.Create<object>(SnippetsRule.NeverInclude, SnippetsRule.AlwaysInclude, SnippetsRule.IncludeAfterTypingIdentifierQuestionTab)).
+                Add(CompletionOptionsStorage.EnterKeyBehavior, ImmutableArray.Create<object>(EnterKeyRule.Never, EnterKeyRule.AfterFullyTypedWord, EnterKeyRule.Always));
```

Comment:
Can we not just use `Enum.GetValues` for these?

Reply:
Use `SnippetsRule` as example here.
https://github.com/dotnet/roslyn/blob/main/src/Features/Core/Portable/Completion/SnippetsRule.cs
It contains four values. And the Default value is `default`.
But it is overridden differently for[ C#](https://github.com/dotnet/roslyn/blob/efdb56597abbe7dff8f5df311f60375877f0b17d/src/Features/CSharp/Portable/Completion/CSharpCompletionService.cs#L55) and [VB ](https://github.com/dotnet/roslyn/blob/efdb56597abbe7dff8f5df311f60375877f0b17d/src/Features/VisualBasic/Portable/Completion/VisualBasicCompletionService.vb#L63) later, 

So, in reality the `default` value is not used, and they are not listed in the unified settings regirstration file.
We just use the real value for each language.
![image](https://github.com/dotnet/roslyn/assets/24360909/62c976c0-7cbd-4074-92ab-d9400863bb09)

So the array here to just to exclude `default` from `Enum.GetValues`

Reply:
Yes, I would prefer to use `Enum.GetValues` and then explicitly remove options that are known to be missing. Each exclusion should have its exclusion reason documented.

Reply:
So I use a similar with like https://github.com/dotnet/roslyn/pull/72946#discussion_r1559449317 to address this.